### ISSUE TITLE
[ARM] can't pair my device / multiple times QNetworkReply::SslHandshakeFailedError

### ISSUE THREAD

**Issue by SupervisedThinking** [moonlight-stream/moonlight-qt#715]
**Describe the bug**
Pairing isn't possible on my (Amlogic) ARM devices while it works on my x86 system.

**Steps to reproduce**
Just try to pair, enter the 4 digit code and it will fail with:
`GeForce Experience returned error: Server certificate mismatch (error 401)`

**Client PC details (please complete the following information)**
https://github.com/SupervisedThinking/LibreELEC-RR/tree/master-rr
https://github.com/moonlight-stream/moonlight-qt/commit/cb8c30db2e4792481ac1d8b3f1a3b19ce9f5aa61
Qt5: [5.15.2+KDE patches]( https://invent.kde.org/qt/qt/qt5/-/tree/5bb3613032c16cb7102d57fec1e4d2265a173748)
SDL 2.0.20
Kernel 5.16.0
OpenSSL 3.0.1
AML S905x

**Server PC details (please complete the following information)**
 - OS: Windows 10
 - GeForce Experience version: 3.24.0.126
 - Nvidia GPU driver: 511.23

**Moonlight Logs (please attach)**
```
vim1:~ # ./moonlight-qt
00:00:00 - Qt Info: Unable to detect Wayland or X11, so EGLFS will be used by default. Set QT_QPA_PLATFORM to override this.
00:00:00 - Qt Info: Setting display mode by default. Set QT_QPA_EGLFS_ALWAYS_SET_MODE=0 to override this.
00:00:00 - SDL Info (0): Compiled with SDL 2.0.20
00:00:00 - SDL Info (0): Running with SDL 2.0.20
00:00:00 - Qt Info: No translation available for  "C"
00:00:02 - Qt Warning: qrc:/gui/main.qml:12:1: QML ApplicationWindow: ToolTip must be attached to an Item
00:00:02 - Qt Info: Found "gamecontrollerdb.txt" at ":/data/gamecontrollerdb.txt"
00:00:02 - SDL Info (0): Loaded 175 new gamepad mappings
00:00:02 - SDL Info (0): V-sync disabled
00:00:02 - SDL Info (0): Sharing DRM FD with SDL
00:00:02 - SDL Info (0): DRM backend supports exporting EGLImage
00:00:02 - SDL Error (0): EGLRenderer: SDL_CreateRenderer() failed: Could not create EGL context (call to eglCreateContext failed, reporting an error of EGL_BAD_CONFIG)
00:00:02 - SDL Info (0): Sharing DRM FD with SDL
00:00:02 - SDL Info (0): Using DRM renderer
00:00:02 - FFmpeg: [h264_mp4toannexb @ 0x23bab70] The input looks like it is Annex B already
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] level=-99
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] Format drm_prime chosen by get_format().
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] avctx requested=181 (drm_prime); get_format requested=181 (drm_prime)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] probing device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] v4l2 capture format not supported
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] Could not find a valid device
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] can't configure decoder
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x23b5320] V4L2 Codec end
00:00:02 - FFmpeg: V4L2 Context destroyed
00:00:02 - SDL Error (0): Unable to open decoder for format: 1
00:00:02 - SDL Info (0): Using SDL renderer
00:00:02 - FFmpeg: [h264_mp4toannexb @ 0x23cf9d0] The input looks like it is Annex B already
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] level=-99
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] Format yuv420p chosen by get_format().
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] avctx requested=0 (yuv420p); get_format requested=0 (yuv420p)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] probing device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] Using device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] requesting formats: output=H264 capture=NM12
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] output: H264 16 buffers initialized: 0256x0144, sizeimage 01620736, bytesperline 00000000
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] Driver 'meson-vdec': Quirks=0x1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] output set status 1074026002 (ON) OK
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] VIDIOC_DECODER_CMD start OK
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output pre VIDIOC_QBUF: index 0, ts=0.000001 count=0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output VIDIOC_QBUF: index 0, ts=0.000001 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] Dq event 5
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] V4L2 capture changed: alloc=0 (256x144) -> (1280x720)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] Source change: SAR: 0/1, crop 0x0 @ 0,0, reinit=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] reinit context
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] capture: NM12 20 buffers initialized: 1280x0720, sizeimage 00983040, bytesperline 00001280
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 0, ts=0.000000 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 1, ts=0.000000 count=2
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 2, ts=0.000000 count=3
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 3, ts=0.000000 count=4
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 4, ts=0.000000 count=5
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 5, ts=0.000000 count=6
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 6, ts=0.000000 count=7
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 7, ts=0.000000 count=8
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 8, ts=0.000000 count=9
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 9, ts=0.000000 count=10
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 10, ts=0.000000 count=11
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 11, ts=0.000000 count=12
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 12, ts=0.000000 count=13
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 13, ts=0.000000 count=14
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 14, ts=0.000000 count=15
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 15, ts=0.000000 count=16
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 16, ts=0.000000 count=17
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 17, ts=0.000000 count=18
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 18, ts=0.000000 count=19
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 19, ts=0.000000 count=20
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] capture set status 1074026002 (ON) OK
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output pre VIDIOC_QBUF: index 0, ts=0.000002 count=0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output VIDIOC_QBUF: index 0, ts=0.000002 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output pre VIDIOC_QBUF: index 0, ts=0.000003 count=0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- output VIDIOC_QBUF: index 0, ts=0.000003 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] capture: Buffer requeue
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] --- capture VIDIOC_QBUF: index 0, ts=0.000001 count=20
00:00:02 - SDL Info (0): Chose SdlRenderer for codec h264_v4l2m2m due to preferred pixel format: 0x0
00:00:02 - SDL Info (0): FFmpeg-based video decoder chosen
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x242d240] V4L2 Codec end
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x242d240] output set status 1074026003 (OFF) OK
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x242d240] capture set status 1074026003 (OFF) OK
00:00:03 - FFmpeg: V4L2 Context destroyed
00:00:03 - Qt Info: Found "gamecontrollerdb.txt" at ":/data/gamecontrollerdb.txt"
00:00:03 - SDL Info (0): Loaded 175 new gamepad mappings
00:00:03 - Qt Info: "X99" is now online at "192.168.1.10:47989"
00:00:03 - SDL Info (0): Captured Qt EGLFS DRM master fd (legacy): 5
00:00:03 - Qt Debug: Current Moonlight version: "3.1.4"
00:00:03 - Qt Warning: Failed to download updated gamepad mappings: QNetworkReply::SslHandshakeFailedError
00:00:03 - Qt Warning: Failed to download latest compatibility data: QNetworkReply::SslHandshakeFailedError
00:00:04 - Qt Info: Discovered mDNS host: "X99.local."
00:00:04 - Qt Info: Resolved "X99.local." to QHostAddress("192.168.1.10")
00:00:04 - Qt Info: Resolved "X99.local." to QHostAddress("fe80::6cfb:cb8a:c352:e163")
00:00:06 - Qt Info: Processing new PC at "192.168.1.10:47989" from mDNS with IPv6 address "<NULL>"
00:00:06 - Qt Info: Executing request: "http://192.168.1.10:47989/serverinfo?uniqueid=0123456789ABCDEF&uuid=466ca4770c0f4a97bbbabaecec466660"
00:00:06 - Qt Info: Executing request: "https://192.168.1.10:47984/serverinfo?uniqueid=0123456789ABCDEF&uuid=905f7875b71c4588aa3b9327ecda5d32"
00:00:06 - Qt Warning: "serverinfo"  request failed with error  QNetworkReply::SslHandshakeFailedError
00:00:06 - Qt Info: Executing request: "http://192.168.1.10:47989/serverinfo?uniqueid=0123456789ABCDEF&uuid=b500bc1ccf7a4fdf9732ce83f38b76c3"
00:00:10 - Qt Info: Pairing with server generation: 7
00:00:10 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=522c81a6f7c74cdf8873caaf31d4e078&devicename=roth&updateState=1&phrase=getservercert&salt=22bc72f22c25858268dca657227440c0&clientcert=2d2d2d2d2d424547494e2043455254494649434154452d2d2d2d2d0a4d494943767a43434161656741774942416749424144414e42676b71686b694739773042415173464144416a4d53457748775944565151444442684f566b6c450a53554567523246745a564e30636d56686253424462476c6c626e51774868634e4d6a45784d4449334d4467314e6a41795768634e4e4445784d4449794d4467310a4e6a4179576a416a4d53457748775944565151444442684f566b6c4553554567523246745a564e30636d56686253424462476c6c626e5177676745694d4130470a4353714753496233445145424151554141344942447741776767454b416f4942415144475153463951524c6a4877386c4a3567495a694a572f69794a457a41790a6d452b395a7779684a4d657539416572666d374d504c376545494f415649347154583051596759673837717939564b32724d795153757a77784565503749324e0a6451395a6e524b707251356775785a3375426a616e777377742b5964516b636b4c714e70417678366c41323157695964362f6e493552784d7a3978416f57476d0a385370627a4b5845456c7742635a694c573155716e353358523167643567652b51394830695569307a5859455a7141384c2f4a4f3965624468314655343468730a5146495362697a3473696f484a4463323041303736314a386d5142346c776b3438646c696a2b714a6b794964586b6944586658484d4c65716f7541304b5676480a62726d4b573578764165425761704b374d363547522b695066394579736156367031464b76576c7738544b7052774a3757543556656f646441674d42414145770a4451594a4b6f5a496876634e4151454c42514144676745424142715a6b4958304e6e69457879344a635a72705772785135624836472f317959483843496c56440a35476c365a45426b445456776d792f743855494d655147447158785471306c2f48753773636f4f336a6d64434c6c36577865636b745461505065616c34426a490a512b39462b69546e55795531617969553243616c436d6635416d667941446a6273596361454c6d6468455a6c396976486d754871594e4d4c47766e7a664750530a7477534d2b434849386234306a4c7465684e573273724d32682b71634e4b374271762b445a314b374e525341637855526266636f5467796d33384855555435740a4c36466f74786d74467a6a5a674c695033554862344e78714233412f7751334c633956643457794a7934316d547641794b2f4e58326b6531454b6a6d6c6a68430a796f4b617559794173643064373255625a686143304e50645432684550586d4d47384c4576435250703045714a684d3d0a2d2d2d2d2d454e442043455254494649434154452d2d2d2d2d0a"
00:00:13 - Qt Warning: "applist"  request failed with error  QNetworkReply::SslHandshakeFailedError
00:00:27 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=f04cb3bd0c1f427c90df67f3386210a7&devicename=roth&updateState=1&clientchallenge=b9324582abe37a3f8842f0daf75e2e7c"
00:00:27 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=07f5d8d7bbed479f80ddf1e8425349af&devicename=roth&updateState=1&serverchallengeresp=f3fbfc7f4452b7cbec8fe3209c53e746a85b1d6da24f0e0a312610482a6fd6c4"
00:00:27 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=c33d3a56bbf34ef59b1a49aa3d938f45&devicename=roth&updateState=1&clientpairingsecret=83f4bd2973c8204d8437770153723b438e72eb2458035ded6e7466871d8bd0b36f578305c9c65a801235f9b45265675eee216a1aaa8aafb05b011fc9cdf112907186b0cda3e4594c038ce14ddf55487e87191d6206eb864ccc9b10f86fc913573d1ad1317e915a8a8a2c26ee51c6cecd53fd20fbdbfea6c01e8fe6d7fc95dacfd5866705d26b20cc6564bbb4e2e9a0a84a9326c8c95890c13bd85a61d301204fc9cb3f6e1ffb68e0f0280393194c562f92adb4f7adca517743ccc5b83a306e588b930424d30817c1eeb696c8e380af154195aad2394750db92f9f0cf9967d225342ce539a6e717e790790e46f0111aa2d281b5ca4f906e25406946ad93dfa415efee275eb85d46ba3d2a2e68540a73c8"
00:00:27 - Qt Info: Executing request: "https://192.168.1.10:47984/pair?uniqueid=0123456789ABCDEF&uuid=e2b523f7694049cf8892a7649dfb86be&devicename=roth&updateState=1&phrase=pairchallenge"
00:00:27 - Qt Warning: "pair"  request failed with error  QNetworkReply::SslHandshakeFailedError
```

**Comment by SupervisedThinking** [comment]
Update: might be related to the cacert 🤔 but then again the question is why it is failing on ARM but not on X86

**Comment by SupervisedThinking** [comment]
- tried a clean install of a clean build
- bumped moonlight to latest
- bumped Qt

```
00:00:00 - SDL Info (0): Compiled with SDL 2.0.20
00:00:00 - SDL Info (0): Running with SDL 2.0.20
00:00:00 - Qt Info: No translation available for  "C"
00:00:02 - Qt Warning: qrc:/gui/main.qml:12:1: QML ApplicationWindow: ToolTip must be attached to an Item
00:00:02 - Qt Info: Found "gamecontrollerdb.txt" at ":/data/gamecontrollerdb.txt"
00:00:02 - SDL Warn (0): 0 new mappings found in gamecontrollerdb.txt. Is it corrupt?
00:00:02 - SDL Info (0): V-sync disabled
00:00:02 - SDL Info (0): Sharing DRM FD with SDL
00:00:02 - SDL Info (0): DRM backend supports exporting EGLImage
00:00:02 - SDL Error (0): EGLRenderer: SDL_CreateRenderer() failed: Could not create EGL context (call to eglCreateContext failed, reporting an error of EGL_BAD_CONFIG)
00:00:02 - SDL Info (0): Sharing DRM FD with SDL
00:00:02 - SDL Info (0): Using DRM renderer
00:00:02 - FFmpeg: [h264_mp4toannexb @ 0x1bb4d50] The input looks like it is Annex B already
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] level=-99
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] Format drm_prime chosen by get_format().
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] avctx requested=181 (drm_prime); get_format requested=181 (drm_prime)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] probing device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] v4l2 capture format not supported
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] Could not find a valid device
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] can't configure decoder
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1b95130] V4L2 Codec end
00:00:02 - FFmpeg: V4L2 Context destroyed
00:00:02 - SDL Error (0): Unable to open decoder for format: 1
00:00:02 - SDL Info (0): Using SDL renderer
00:00:02 - FFmpeg: [h264_mp4toannexb @ 0x1bc5df0] The input looks like it is Annex B already
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] level=-99
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] Format yuv420p chosen by get_format().
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] avctx requested=0 (yuv420p); get_format requested=0 (yuv420p)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] probing device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] Using device /dev/video0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] driver 'meson-vdec' on card 'Amlogic Video Decoder' in mplane mode
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] requesting formats: output=H264 capture=NM12
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] output: H264 16 buffers initialized: 0256x0144, sizeimage 01620736, bytesperline 00000000
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] Driver 'meson-vdec': Quirks=0x1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] output set status 1074026002 (ON) OK
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] VIDIOC_DECODER_CMD start OK
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output pre VIDIOC_QBUF: index 0, ts=0.000001 count=0
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output VIDIOC_QBUF: index 0, ts=0.000001 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] Dq event 5
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] V4L2 capture changed: alloc=0 (256x144) -> (1280x720)
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] Source change: SAR: 0/1, crop 0x0 @ 0,0, reinit=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] reinit context
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] capture: NM12 20 buffers initialized: 1280x0720, sizeimage 00983040, bytesperline 00001280
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 0, ts=0.000000 count=1
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 1, ts=0.000000 count=2
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 2, ts=0.000000 count=3
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 3, ts=0.000000 count=4
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 4, ts=0.000000 count=5
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 5, ts=0.000000 count=6
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 6, ts=0.000000 count=7
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 7, ts=0.000000 count=8
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 8, ts=0.000000 count=9
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 9, ts=0.000000 count=10
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 10, ts=0.000000 count=11
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 11, ts=0.000000 count=12
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 12, ts=0.000000 count=13
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 13, ts=0.000000 count=14
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 14, ts=0.000000 count=15
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 15, ts=0.000000 count=16
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 16, ts=0.000000 count=17
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 17, ts=0.000000 count=18
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 18, ts=0.000000 count=19
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 19, ts=0.000000 count=20
00:00:02 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] capture set status 1074026002 (ON) OK
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output pre VIDIOC_QBUF: index 0, ts=0.000002 count=0
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output VIDIOC_QBUF: index 0, ts=0.000002 count=1
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output pre VIDIOC_QBUF: index 0, ts=0.000003 count=0
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- output VIDIOC_QBUF: index 0, ts=0.000003 count=1
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] capture: Buffer requeue
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] --- capture VIDIOC_QBUF: index 0, ts=0.000001 count=20
00:00:03 - SDL Info (0): Chose SdlRenderer for codec h264_v4l2m2m due to preferred pixel format: 0x0
00:00:03 - SDL Info (0): FFmpeg-based video decoder chosen
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] V4L2 Codec end
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] output set status 1074026003 (OFF) OK
00:00:03 - FFmpeg: [h264_v4l2m2m @ 0x1bba390] capture set status 1074026003 (OFF) OK
00:00:03 - FFmpeg: V4L2 Context destroyed
00:00:03 - Qt Info: Found "gamecontrollerdb.txt" at ":/data/gamecontrollerdb.txt"
00:00:03 - SDL Warn (0): 0 new mappings found in gamecontrollerdb.txt. Is it corrupt?
00:00:04 - SDL Info (0): Captured Qt EGLFS DRM master fd (legacy): 5
00:00:04 - Qt Debug: Current Moonlight version: "3.2.0"
00:00:04 - Qt Warning: Failed to download updated gamepad mappings: QNetworkReply::SslHandshakeFailedError
00:00:04 - Qt Warning: Failed to download latest compatibility data: QNetworkReply::SslHandshakeFailedError
00:00:04 - Qt Info: Discovered mDNS host: "X99.local."
00:00:04 - Qt Info: Resolved "X99.local." to QHostAddress("192.168.1.10")
00:00:04 - Qt Info: Resolved "X99.local." to QHostAddress("fe80::6cfb:cb8a:c352:e163")
00:00:06 - Qt Info: Processing new PC at "192.168.1.10:47989" from mDNS with IPv6 address "<NULL>"
00:00:06 - Qt Info: Executing request: "http://192.168.1.10:47989/serverinfo?uniqueid=0123456789ABCDEF&uuid=a18fe4700aac43158cc65139f453146b"
00:00:27 - Qt Info: Pairing with server generation: 7
00:00:27 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=b549324f17454fd283bd7430e9bd3417&devicename=roth&updateState=1&phrase=getservercert&salt=747940e18d074f7dbea6887527c9023a&clientcert=2d2d2d2d2d424547494e2043455254494649434154452d2d2d2d2d0a4d494943767a43434161656741774942416749424144414e42676b71686b694739773042415173464144416a4d53457748775944565151444442684f566b6c450a53554567523246745a564e30636d56686253424462476c6c626e51774868634e4d6a49774d5449784d446b314d6a55785768634e4e4449774d5445324d446b310a4d6a5578576a416a4d53457748775944565151444442684f566b6c4553554567523246745a564e30636d56686253424462476c6c626e5177676745694d4130470a4353714753496233445145424151554141344942447741776767454b416f494241514362736147325369365145625670446d36456c694147723550483972734b0a2b63657334756277574b537a4d7945713455314a366e4739526b317a642f4c4f6866536f41752b35745a423642653336474644382b426649454178635a416a520a686d533655574a502b386c653065326645506e4330347136426a6f2b76356769654157744a36495947353063504b416277534d4b41317267796f74506c6e652b0a7876674630593437693347614e58757853496861695330544d764269745768627752375674746f424f4a4e5a75705a41685133434d524e4c534e6133467372410a4b5a7a4f365657426f7a6e5a6941314d7278772f424879343334797a69563662344a314f30525a503744694d61694a324f4867394d6f6f426d4b784a654265440a583064504b336f3177682b6e705a55417a5633653041394e32545272456b494d4d4b7878663044794d5476477a3930533076676d504c4a7641674d42414145770a4451594a4b6f5a496876634e4151454c42514144676745424145726754726e4148686a7953685a71526241434a3770673551433369544159685a6361386b75570a6c5a516e634e4c6a566437664662574c744f387243436f424c6f73575139453535706f6b7a767446776d69427a316c506d6178557737444936732f756d4a72520a425055484135432f734f4962374f6f422f556e756e75717330684836614932304d77386a674d434b74576c7551796835537a7156376647307a5443674764346a0a5347623835703030536d77346c306344537a746835316c7a4f334f4c734f6b4d6b622b316966307044583149303645644f3368697453583938747a55796255710a4530376753497875666b6a7a367469793947786672654338472f74456965516f763634417731525a6530793331323934787a74337856634674657045523441700a4a42636433726a7675656769547a6c4f5a45334e78563654445874714d71336b5a437654356c366e4256567565454d3d0a2d2d2d2d2d454e442043455254494649434154452d2d2d2d2d0a"
00:00:36 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=338f449328f5400db81725bdf7ee031e&devicename=roth&updateState=1&clientchallenge=83c9ed6ed169a394a2eea8935ecadb8d"
00:00:36 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=2e2ec65ea30347679b9d6f89790ea051&devicename=roth&updateState=1&serverchallengeresp=60327269a68ab045ae1715f4a02446b6aede652996484c2b41c9fe4b0352a0d8"
00:00:36 - Qt Info: Executing request: "http://192.168.1.10:47989/pair?uniqueid=0123456789ABCDEF&uuid=fc1abc6a84d14fbc8bbe344c8ee7fc5e&devicename=roth&updateState=1&clientpairingsecret=5268d662f96aa8da6b445ac466cc1622363ed2a7426f9b8b48a210cd12535f1a2d160089db566c7b82c4ce35dd6cb246f4011a734ec17f129503879e0c16a6b7aeca993724bca6f5b8f3273769583f60d32d88304b3115086cbb2c16be54e5371a51458a76eec6546a2b2230825d123b089bd49c69dd846cf0ec344be6addbbd253eafe78278d9696cd3c06117f4e9c8537a433e5aa0c521f1e28914f6b54955689e63a86f723ed79c7a183c63d5463d6891e4f6eb07364bcf85b679e9a38f478354fcedab4be8b1a25cd20b74e4bd5a2493f625460e2b869da6e6de5763b0f33dab651970efdd86fcab9f83d48e116e2d61c2dc7a252399495e2cb17afc19d5b86a83d6973aeed8a3efdf423a7abbf5"
00:00:37 - Qt Info: Executing request: "https://192.168.1.10:47984/pair?uniqueid=0123456789ABCDEF&uuid=bc3fa66844aa45ffbefd5bcd6ff3f9d2&devicename=roth&updateState=1&phrase=pairchallenge"
00:00:37 - Qt Warning: "pair"  request failed with error  QNetworkReply::SslHandshakeFailedError
```

**Comment by cgutman** [comment]
I think you have an incompatibility between your Qt and OpenSSL version.

All of your HTTPS connections are failing, including normal ones to sites with certificates issued by real CAs:
```
00:00:04 - Qt Warning: Failed to download updated gamepad mappings: QNetworkReply::SslHandshakeFailedError
00:00:04 - Qt Warning: Failed to download latest compatibility data: QNetworkReply::SslHandshakeFailedError
```

If you can resolve that, your pairing issue will probably go away too.

**Comment by SupervisedThinking** [comment]
Yes that's what I've had in mind too - I've tested the latest NV driver with an old image & it works so I guess this won't be the problem. But here is where the fun begins: the same package versions work fine on my x86 build while they fail for ARM.

My suspicion is that OpenSSL 3.0.y does not work reliable on ARM or Qt5 + KDE patches fails. I've already checked you code and found the url which fail e.g. `https://moonlight-stream.org/SDL_GameControllerDB/gamecontrollerdb.txt` but when I used [this](https://docs.pingidentity.com/bundle/solution-guides/page/iqs1569423823079.html) to test the ssl connectifity to your domain it worked on my ARM boards 🤔 

Next step will be reverting the OpenSSL bump and report back.
